lib/ansible/modules/cloud/amazon/aws_codedeploy_application.py

```diff
@@ -0,0 +1,433 @@
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_codedeploy_application(client, name, module):
+    """Get the details of the CodeDeploy application."""
+    try:
+        return client.get_application(applicationName=name)
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'ApplicationDoesNotExistException':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+    return None
+
+
+def get_codedeploy_application_deployment_group(client, name, dg, module):
+    """Get the details of the CodeDeploy deployment group."""
+    try:
+        return client.get_deployment_group(applicationName=name, deploymentGroupName=dg)
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'ApplicationDoesNotExistException':
+            return None
+        if e.response['Error']['Code'] == 'DeploymentGroupDoesNotExistException':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+    return None
+
+
+def create_codedeploy_deployment_group(client, name, dg, dc, ec2_tf, opf, asg, sra, tc, ac, arc, ds, bgdc, lbi, module):
+    """Create a CodeDeploy deployment group for the application."""
+    changed = False
+    try:
+        dg_params = dict()
+        if dc is not None:
+            dg_params['deploymentConfigName'] = dc
+        if ec2_tf is not None:
+            dg_params['ec2TagFilters'] = ec2_tf
+        if opf is not None:
+            dg_params['onPremisesInstanceTagFilters'] = opf
+        if asg is not None:
+            dg_params['autoScalingGroups'] = asg
+        if tc is not None:
+            dg_params['triggerConfigurations'] = tc
+        if ac is not None:
+            dg_params['alarmConfiguration'] = ac
+        if arc is not None:
+            dg_params['autoRollbackConfiguration'] = arc
+        if ds is not None:
+            dg_params['deploymentStyle'] = ds
+        if bgdc is not None:
+            dg_params['blueGreenDeploymentConfiguration'] = bgdc
+        if lbi is not None:
+            dg_params['loadBalancerInfo'] = lbi
+        response = client.create_deployment_group(
+            applicationName=name,
+            deploymentGroupName=dg,
+            serviceRoleArn=sra,
+            **dg_params)
+        changed = True
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'ApplicationDoesNotExistException':
+            return None
+        if e.response['Error']['Code'] == 'DeploymentGroupDoesNotExistException':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+    return changed
+
+
+def create_codedeploy_application(client, name, dg, ndg, dc, ec2_tf, opf, asg, sra, tc, ac, arc, ds, bgdc, lbi, module):
```

Comment:
Please use named parameters with parameter lists this long, and without abbreviations which makes is much harder to read.